thunder/tests/test_jit_general.py

```diff
@@ -1439,3 +1439,47 @@ def foo(a):
     assert_close(actual, expected)
     assert thunder.cache_misses(jfoo) == 1
     assert thunder.cache_hits(jfoo) == 1
+
+
+@pytest.mark.parametrize(
+    "device",
+    ("cpu", "cuda"),
+)
+def test_cache_symbolic_values_reshape_numel(device):
+    if device == "cuda" and not torch.cuda.is_available():
+        pytest.skip("CUDA not available")
+
+    def foo(a):
+        a = torch.reshape(a, [a.numel()])
+        return a.relu()
+
+    jfoo = thunder.jit(foo, cache="symbolic values")
+
+    a = torch.randn(2, 3, 8, requires_grad=True, device=device)
```

Comment:
nit: is the `requires_grad` critical here? seems like we'd still need to compute the shape even w/o grad, right?

Reply:
Yes you are right.

This is just for test coverage, with `requires_grad`, we'll run grad transform on top of the original trace, which is where things tend to break. 😢

Reply:
ack, okay